rotkehlchen/api/v1/schemas.py

```diff
@@ -3694,3 +3701,54 @@ def make_calendar_query(
             identifiers=data['identifiers'],
         )
         return {'filter_query': filter_query}
+
+
+class CalendarReminderCommonEntrySchema(Schema):
+    AT_TIME_REGEX = re.compile(r'^(2[0-3]|[0-1][0-9]):[0-5][0-9]$')
+    event_id = fields.Integer(
+        required=True,
+        validate=webargs.validate.Range(min=1, error='event_id has to be bigger than 0'),
+    )
+    amount = fields.Integer(required=True)
+    unit = fields.String(
+        required=True,
+        validate=webargs.validate.OneOf(choices=typing.get_args(VALID_REMINDER_UNITS)),
+    )
+    at_time = fields.String(load_default=None)
+
+    @validates_schema
+    def validate_schema(
+            self,
+            data: dict[str, Any],
+            **_kwargs: Any,
+    ) -> None:
+        if data['at_time'] is None:
+            return
+
+        if not self.AT_TIME_REGEX.match(data['at_time']):
+            raise ValidationError(
+                message='Invalid notification time provided',
+                field_name='at_time',
+            )
+
+
+class NewCalendarReminderSchema(CalendarReminderCommonEntrySchema):
+    @post_load
+    def make_calendar_entry(self, data: dict[str, Any], **_kwargs: dict[str, Any]) -> dict[str, Any]:  # noqa: E501
+        return {
+            'reminder': ReminderEntry(
+                identifier=data.get('identifier', 0),
+                event_id=data['event_id'],
+                amount=data['amount'],
+                unit=data['unit'],
+                at_time=data['at_time'],
+            ),
+        }
+
+
+class UpdateCalendarReminderSchema(NewCalendarReminderSchema, IntegerIdentifierSchema):
+    ...
+
+
+class QueryCalendarReminderSchema(Schema):
```

Comment:
Don't use this. Use `IntegerIdentifierSchema`. Don't remake multiple schema classes for the same thing but change the name.